specs/electra/beacon-chain.md

```diff
@@ -1070,6 +1043,7 @@ def process_execution_payload(state: BeaconState, body: BeaconBlockBody, executi
     assert execution_engine.verify_and_notify_new_payload(
```

Comment:
Is the definition of `verify_and_notify_new_payload()` and `notify_new_payload()` intentionally left out for another PR or coordination with execution-api side of things?

Would need to update these two functions if not.

Reply:
@ensi321 Do you mean doing the following:
- update function `notify_new_payload` to receive `execution_requests` as a parameter
- update function `verify_and_notify_new_payload` to correctly pass `execution_requests` as a parameter parameter to `notify_new_payload`?

Reply:
IMO, this PR should update `notify_new_payload` call, but there is no need to update Engine API as requests are already incorporated into `newPayloadV4` definition

Reply:
@ensi321 and @mkalinin I have made the changes. Let me know what you think.

Reply:
> IMO, this PR should update `notify_new_payload` call, but there is no need to update Engine API as requests are already incorporated into `newPayloadV4` definition

As far as I can tell from the two proposals in engine API ethereum/execution-apis#577 ethereum/execution-apis#565, `requests` indeed is part of the `ExecutionPayloadV4` definition. 

But this PR is proposing to remove the 3 requests field from `ExecutionPayload` and store them in `BeaconBlockBody` under `ExecutionRequests`.

So I think we need to update engine API spec to either
1) Add an extra param in `engine_newPayloadV4` for `ExecutionRequests`
2) CL populates `ExecutionPayloadV4.requests` with `ExecutionRequests` before calling `engine_newPayloadV4`

Correct me if I am wrong

Reply:
The design goal of `notify_new_payload` is to make the abstraction layer in the CL spec that would represent the engine. CL spec is agnostic to the specific implementation of the engine, in our case Engine API. In other words, Engine API requests aren’t needed to be reflected in the CL spec, and the spec should rather operate with the structures defined in it. So, `ExecutionRequests` sent to `notify_new_payload` as a separate parameter and sit in the `requests` field of the `ExecutionPayloadV4` looks fine